Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -208,15 +208,13 @@ public struct PackageBuilder {
     ///   - includingTestModules: Whether the package's test modules should be loaded.
     public func construct(includingTestModules: Bool) throws -> Package {
         let modules = try constructModules()
-        let testModules: [Module]
-        if includingTestModules {
-            testModules = try constructTestModules(modules: modules)
-        } else {
-            testModules = []
-        }
+        // It is unfortunate that test modules has to be constructed even when they are
+        // not requested by the clients but they might be referenced in the manifest file
+        // so that needs to be validated.
+        let testModules = try constructTestModules(modules: modules)
```

Comment:
I think it is ok that we construct test modules here, the performance cost should be negligible and I think it is actually better to not have the graph structure change depending on the use case.